01 OperatorOverloading/OperatorOverloading.Model/Money.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+
+namespace OperatorOverloading.Model
+{
+    public class Money
+    {
+        private double _amount;
+        private string _currency;
+
+        public double Amount
+        {
+            get { return _amount; }
+            private set
+            {
+                if (value < 0)
+                {
+                    throw new InvalidCurrencyException(Messages.AmountNegative);
+                }
+                if (double.IsPositiveInfinity(value))
+                {
+                    throw new InvalidCurrencyException(Messages.AmountTooLarge);
+                }
+                _amount = value;
+            }
+        }
+
+        public string Currency
+        {
+            get { return _currency; }
+            private set
+            {
+                if (string.IsNullOrEmpty(value))
+                {
+                    throw new InvalidCurrencyException(Messages.InvalidCurrency);
+                }
+                _currency = value;
+            }
+        }
+
+        public Money(double amt, string cur)
+        {
+            Amount = amt;
+            Currency = cur;
+        }
+
+        public Money(string inputAmount)
+        {
+            string[] amountArr;
+            double amt;
+
+            if (string.IsNullOrEmpty(inputAmount))
+            {
+                throw new InvalidCurrencyException(Messages.NullValue);
+            }
+
+            amountArr = inputAmount.Split(' ');
```

Comment:
Change to `var amountArr = inputAmount.Split(' ');` Also define amount after the null check.